contrib/platform/src/com/sun/jna/platform/win32/IPHlpAPI.java

```diff
@@ -0,0 +1,280 @@
+    int IF_MAX_STRING_SIZE = 256;
+    int IF_MAX_PHYS_ADDRESS_LENGTH = 32;
+    int MAX_INTERFACE_NAME_LEN = 256;
+    int MAXLEN_IFDESCR = 256;
+    int MAXLEN_PHYSADDR = 8;
+    int MAX_HOSTNAME_LEN = 128;
+    int MAX_DOMAIN_NAME_LEN = 128;
+    int MAX_SCOPE_ID_LEN = 256;
+
+    /**
+     * The MIB_IFROW structure stores information about a particular interface.
+     * 
+     * @see <A HREF=
+     *      "https://docs.microsoft.com/en-us/previous-versions/windows/desktop/api/ifmib/ns-ifmib-_mib_ifrow">MIB_IFROW</A>
+     */
+    @FieldOrder({ "wszName", "dwIndex", "dwType", "dwMtu", "dwSpeed", "dwPhysAddrLen", "bPhysAddr", "dwAdminStatus",
+            "dwOperStatus", "dwLastChange", "dwInOctets", "dwInUcastPkts", "dwInNUcastPkts", "dwInDiscards",
+            "dwInErrors", "dwInUnknownProtos", "dwOutOctets", "dwOutUcastPkts", "dwOutNUcastPkts", "dwOutDiscards",
+            "dwOutErrors", "dwOutQLen", "dwDescrLen", "bDescr" })
+    class MIB_IFROW extends Structure {
+        public char[] wszName = new char[MAX_INTERFACE_NAME_LEN];
+        public int dwIndex;
+        public int dwType;
+        public int dwMtu;
+        public int dwSpeed;
+        public int dwPhysAddrLen;
+        public byte[] bPhysAddr = new byte[MAXLEN_PHYSADDR];
+        public int dwAdminStatus;
+        public int dwOperStatus;
+        public int dwLastChange;
+        public int dwInOctets;
+        public int dwInUcastPkts;
+        public int dwInNUcastPkts;
+        public int dwInDiscards;
+        public int dwInErrors;
+        public int dwInUnknownProtos;
+        public int dwOutOctets;
+        public int dwOutUcastPkts;
+        public int dwOutNUcastPkts;
+        public int dwOutDiscards;
+        public int dwOutErrors;
+        public int dwOutQLen;
+        public int dwDescrLen;
+        public byte[] bDescr = new byte[MAXLEN_IFDESCR];
+    }
+
+    /**
+     * The MIB_IF_ROW2 structure stores information about a particular
+     * interface.
+     * 
+     * @see <A HREF=
+     *      "https://msdn.microsoft.com/library/windows/hardware/ff559214">MIB_IF_ROW2</A>
+     */
+    @FieldOrder({ "InterfaceLuid", "InterfaceIndex", "InterfaceGuid", "Alias", "Description", "PhysicalAddressLength",
+            "PhysicalAddress", "PermanentPhysicalAddress", "Mtu", "Type", "TunnelType", "MediaType",
+            "PhysicalMediumType", "AccessType", "DirectionType", "InterfaceAndOperStatusFlags", "OperStatus",
+            "AdminStatus", "MediaConnectState", "NetworkGuid", "ConnectionType", "TransmitLinkSpeed",
+            "ReceiveLinkSpeed", "InOctets", "InUcastPkts", "InNUcastPkts", "InDiscards", "InErrors", "InUnknownProtos",
+            "InUcastOctets", "InMulticastOctets", "InBroadcastOctets", "OutOctets", "OutUcastPkts", "OutNUcastPkts",
+            "OutDiscards", "OutErrors", "OutUcastOctets", "OutMulticastOctets", "OutBroadcastOctets", "OutQLen" })
+    class MIB_IF_ROW2 extends Structure {
+        public long InterfaceLuid; // 64-bit union

```

Comment:
Is this safe to use the raw value here instead of the union?

Reply:
It's safe, as it's 64-bits.   The alternative would be a multi-layer NET_LUID union in JNA with one alternative being a 64-bit Value (the intended return) and the actual information encoded in 5 of the 8 bytes.  See https://msdn.microsoft.com/library/windows/hardware/ff568747 .  Happy to try to add that NET_LUID structure if I can figure out how; but at this layer it's possible for it to be returned and then parsed by the user with shifting and bitmasks.

Reply:
I agree, retrieving the correct value can be implemented as a helper method on the structure, if needed.